lib/datadog/statsd/uds_connection.rb

```diff
@@ -28,6 +28,9 @@ def send_message(message)
         socket.sendmsg_nonblock(message)
       rescue Errno::ECONNREFUSED, Errno::ECONNRESET, Errno::ENOENT => e
         @socket = nil
+        # TODO: FIXME: This error should be considered as a retryable error in the
```

Comment:
I found this bug, it was not caught by the previous tests: If we catch a connection refused or reset, we don't try to reconnect to the UDS socket. For the missing socket, it seems relevant, it means that the socket was never found, but for other cases, I think we should retry.